### PR TITLE
build: regenerate api routes and resolve merge error in catchup

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

Somehow the [latest merge](https://github.com/algorand/go-algorand/actions/runs/6799752907) caused codegen checker error, this regenerates stuff.

Regenerated with
```sh
touch daemon/algod/api/algod.oas2.json
make -C daemon/algod/api generate
```

Also there are two conflicting catchup PRs were merged causing [error usage ](https://app.circleci.com/pipelines/github/algorand/go-algorand/17170/workflows/34b8d2e6-dfc0-48fd-89de-43bc4e374c34/jobs/258888)that was changed. Fixed as well.

### CODE REVIEW

File: daemon/algod/api/algod.oas3.yml [hunk 1]
```diff
@@ -7192,5 +7192,6 @@
     {
       "name": "private"
     }
-  ]
+  ],
+  "x-original-swagger-version": "2.0"
```
Comment (winder) [inline daemon/algod/api/algod.oas3.yml:7196]: This file is generated by a webapp, I guess that app changed the output to include this line.

**Review by winder — APPROVED** [review]

File: daemon/algod/api/algod.oas3.yml (hunk 1 above)
Comment (algorandskiy) [reply daemon/algod/api/algod.oas3.yml:7196]: ah `curl https://converter.swagger.io/api/convert` call, thanks for clarification.

File: catchup/service.go [hunk 2]
```diff
@@ -310,7 +310,8 @@ func (s *Service) fetchAndWrite(ctx context.Context, r basics.Round, prevFetchCo
 				return false
 			}
 			failureRank := peerRankDownloadFailed
-			if err == errNoBlockForRound {
+			var nbfe noBlockForRoundError
+			if errors.As(err, &nbfe) {
```
Comment (winder) [inline catchup/service.go:314]: This looks like it was meant for a different PR

Comment (algorandskiy) [reply catchup/service.go:314]: I guess these both fix master but yes, has different cause

**Review by winder — APPROVED** [review]
All of these changes look fine. One of them doesn't match the current description/title.

**Review by jannotti — APPROVED** [review]

**Pull request merged**